### PR TITLE
show in url also product dir if it is defined

### PR DESCRIPTION
**Pull request by jreidinger**
snimek obrazovky_2017-09-13_11-00-46

zypper lr output:
```
Repository priorities are without effect. All enabled repositories share the same priority.

# | Alias                    | Name                               | Enabled | GPG Check | Refresh
--+--------------------------+------------------------------------+---------+-----------+--------
1 | SUSE_SLE-15_GA           | Code 15 SP0 GA platform (standard) | Yes     | (r ) Yes  | No     
2 | leanos15-15-0            | leanos15-15-0                      | Yes     | (r ) Yes  | No     
3 | multi_modules_Basesystem | multi modules (/Basesystem)        | Yes     | (r ) Yes  | Yes    
4 | multi_modules_Legacy     | multi modules (/Legacy)            | Yes     | (r ) Yes  | Yes    
```

### CODE REVIEW

**Comment by lslezak** [comment]
Looks OK, thanks!

**Comment by lslezak** [comment]
And what is the alias in this case? What does `zypper lr` display?

The alias is created from the name later (https://github.com/yast/yast-packager/pull/278/files#diff-3181dd45a1dd0f7bbc892e69925b0989R211), I'm not sure how the slash (`/`) in the alias is handled. Alias is used for the `*.repo` file name is it can be a bit tricky...

**Comment by jreidinger** [comment]
@lslezak this is output

```
Repository priorities are without effect. All enabled repositories share the same priority.

# | Alias                       | Name                               | Enabled | GPG Check | Refresh
--+-----------------------------+------------------------------------+---------+-----------+--------
1 | SUSE_SLE-15_GA              | Code 15 SP0 GA platform (standard) | Yes     | (r ) Yes  | No     
2 | leanos15-15-0               | leanos15-15-0                      | Yes     | (r ) Yes  | No     
3 | multi_product_(/Basesystem) | multi product (/Basesystem)        | Yes     | (r ) Yes  | Yes    
4 | multi_product_(/Legacy)     | multi product (/Legacy)            | Yes     | (r ) Yes  | Yes 
```

maybe we can strip there at least `()`

**Comment by lslezak** [comment]
I'd remove all `()/`, using these on command line requires escaping, that's ugly.

**Comment by jreidinger** [comment]
@lslezak in general or only in this case? ( multi product repo )

**Comment by lslezak** [comment]
In general, those characters should not be used for any repository. So remove it in all cases.

File: src/include/packager/repositories_include.rb [hunk 1]
```diff
@@ -204,8 +207,8 @@ def createSourceImpl(url, plaindir, download, preffered_name, force_alias)
           end
           alias_name = ""
           if force_alias == ""
-            # replace " " -> "_" (avoid spaces in .repo file name)
-            alias_name = Builtins.mergestring(Builtins.splitstring(name, " "), "_")
+            # replace " " -> "_" (avoid spaces in .repo file name), and remove shell unfriendly '/()'
```
Comment (lslezak) [inline src/include/packager/repositories_include.rb]: NP: Maybe we could remove more characters like `*&<>'"`...

File: src/include/packager/repositories_include.rb [hunk 2]
```diff
@@ -204,8 +210,8 @@ def createSourceImpl(url, plaindir, download, preffered_name, force_alias)
           end
           alias_name = ""
           if force_alias == ""
-            # replace " " -> "_" (avoid spaces in .repo file name)
-            alias_name = Builtins.mergestring(Builtins.splitstring(name, " "), "_")
+            # replace " " -> "_" (avoid spaces in .repo file name) and remove shell unfriendly '/()'
```
Comment (lslezak) [inline src/include/packager/repositories_include.rb]: NP: Update the comment as well, `remove shell unfriendly characters` sounds good enough.

File: package/yast2-packager.changes [hunk 3]
```diff
@@ -1,4 +1,12 @@
 -------------------------------------------------------------------
+Thu Sep 14 14:30:50 UTC 2017 - jreidinger@suse.com
+
+- show product path in url for multi product medias
```
Comment (lslezak) [inline package/yast2-packager.changes]: Bad English: singular is `medium`, plural is `media`
(Fix the line below as well.)

**Pull request merged**